### PR TITLE
fix: enableCSPNonces behavior

### PR DESCRIPTION
**Pull request by climba03003**
Resolves: #118
Resolves: #119
Resolves: #120

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by climba03003** [comment]
@katywings Can you review this PR? Is it fixed with all the three issue you raised?

File: test.js [hunk 1]
```diff
@@ -192,7 +192,112 @@ test('allow merging options for enableCSPNonces', async (t) => {
     t.ok(cspCache.script)
     t.ok(cspCache.style)
     t.includes(res.headers, {
-      'content-security-policy': `default-src 'self';script-src 'self' nonce-${cspCache.script};style-src 'self' nonce-${cspCache.style}`
+      'content-security-policy': `default-src 'self';script-src 'self' 'nonce-${cspCache.script}';style-src 'self' 'nonce-${cspCache.style}'`
+    })
+  } catch (err) {
+    t.error(err)
+  }
```
Comment (mcollina) [inline test.js:199]: the try-catch wrapper is not needed, remove it

File: test.js [hunk 2]
```diff
@@ -192,7 +192,112 @@ test('allow merging options for enableCSPNonces', async (t) => {
...
+    res = await fastify.inject({ method: 'GET', url: '/' })
+    cspCache = res.json()
+    t.ok(cspCache.script)
+    t.ok(cspCache.style)
+    t.includes(res.headers, {
+      'content-security-policy': `default-src 'self';script-src 'self' 'nonce-${cspCache.script}';style-src 'self' 'nonce-${cspCache.style}'`
+    })
+  } catch (err) {
+    t.error(err)
+  }
```
Comment (mcollina) [inline test.js:240]: remove the try-catch wrapper

**Review by mcollina — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by katywings — APPROVED** [review]
Thank you so much for this 😍! Lookin great 🤩

**Pull request merged**